### PR TITLE
Add documentation for the document registry and manager

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

**Comment by blink1073** [comment]
I'm done, will squash the commits after review.

File: tutorial/documents.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+for NotebookPanel widgets.  Another example would be to add a button to the 
+toolbar of each NotebookPanel widget.
+
+## File Types 
+Intended to be used in a "Create New" dialog, providing a list of known 
+file types.
+
+## File Creators 
+Intended for create quick launch file creators. The default use will be for 
+the "create new" dropdown in the file browser, giving list of items that can
```
Comment (afshin) [inline tutorial/documents.md]: "Create New" is how it's capitalized above, so whichever is the correct case should be used.

File: tutorial/documents.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+## Document Model
+Created by the model factories and passed to widget factories and widget 
+extensions.
+
+## Document Context
+Created by the Document Manager and passed to widget factories and
+widget extensions.  They are used to provide an abstracted interface
+to the session and contents API from jupyter-js-services for the 
+given model.  They are tied to a model can be shared between widgets.
```
Comment (afshin) [inline tutorial/documents.md]: > They are tied to a model *and* can be shared between widgets.

File: tutorial/documents.md (hunk 1 above)
Comment (blink1073) [inline tutorial/documents.md]: The implication is that one is a title and one is a button with a plus sign and a dropdown.

**Pull request merged**